[I 130910 14:45:31 app:9] app start
[I 130910 15:53:18 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 15:53:18 app:9] app start
[I 130910 15:53:30 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 15:53:30 app:9] app start
[I 130910 15:53:35 web:1635] 200 GET / (192.168.10.148) 6.79ms
[I 130910 15:53:35 web:1635] 304 GET /favicon.ico (192.168.10.148) 5.55ms
[I 130910 15:53:35 web:1635] 304 GET / (192.168.10.148) 1.32ms
[I 130910 15:53:42 web:1635] 200 POST / (192.168.10.148) 3.29ms
[I 130910 16:07:52 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 16:07:52 app:9] app start
[I 130910 16:07:54 autoreload:186] /home/jack/workspace/order/settings.py modified; restarting server
[I 130910 16:07:54 app:9] app start
[I 130910 16:08:06 web:1635] 200 GET / (192.168.10.148) 5.55ms
[E 130910 16:08:14 web:1228] Uncaught exception POST / (192.168.10.148)
    HTTPRequest(protocol='http', host='192.168.10.105:8888', method='POST', uri='/', version='HTTP/1.1', remote_ip='192.168.10.148', headers={'Origin': 'http://192.168.10.105:8888', 'Content-Length': '32', 'Accept-Language': 'zh-CN,zh;q=0.8', 'Accept-Encoding': 'gzip,deflate,sdch', 'Host': '192.168.10.105:8888', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'User-Agent': 'Mozilla/5.0 (Windows NT 5.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/30.0.1581.2 Safari/537.36', 'Connection': 'keep-alive', 'Referer': 'http://192.168.10.105:8888/', 'Cache-Control': 'max-age=0', 'Content-Type': 'application/x-www-form-urlencoded'})
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1141, in _when_complete
        callback()
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1162, in _execute_method
        self._when_complete(method(*self.path_args, **self.path_kwargs),
      File "/home/jack/workspace/order/handlers/qhandler.py", line 44, in post
        file_name = allset.data_dir + date_str
    UnboundLocalError: local variable 'date_str' referenced before assignment
[E 130910 16:08:14 web:1635] 500 POST / (192.168.10.148) 62.71ms
[I 130910 16:08:50 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 16:08:51 app:9] app start
[I 130910 16:08:53 web:1635] 304 GET / (192.168.10.148) 5.79ms
[E 130910 16:09:00 web:1228] Uncaught exception POST / (192.168.10.148)
    HTTPRequest(protocol='http', host='192.168.10.105:8888', method='POST', uri='/', version='HTTP/1.1', remote_ip='192.168.10.148', headers={'Origin': 'http://192.168.10.105:8888', 'Content-Length': '32', 'Accept-Language': 'zh-CN,zh;q=0.8', 'Accept-Encoding': 'gzip,deflate,sdch', 'Host': '192.168.10.105:8888', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'User-Agent': 'Mozilla/5.0 (Windows NT 5.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/30.0.1581.2 Safari/537.36', 'Connection': 'keep-alive', 'Referer': 'http://192.168.10.105:8888/', 'Cache-Control': 'max-age=0', 'Content-Type': 'application/x-www-form-urlencoded'})
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1141, in _when_complete
        callback()
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1162, in _execute_method
        self._when_complete(method(*self.path_args, **self.path_kwargs),
      File "/home/jack/workspace/order/handlers/qhandler.py", line 54, in post
        fd.open(file_name)
    UnboundLocalError: local variable 'fd' referenced before assignment
[E 130910 16:09:00 web:1635] 500 POST / (192.168.10.148) 11.64ms
[I 130910 16:09:28 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 16:09:28 app:9] app start
[I 130910 16:09:32 web:1635] 304 GET / (192.168.10.148) 4.96ms
[E 130910 16:09:36 web:1228] Uncaught exception POST / (192.168.10.148)
    HTTPRequest(protocol='http', host='192.168.10.105:8888', method='POST', uri='/', version='HTTP/1.1', remote_ip='192.168.10.148', headers={'Origin': 'http://192.168.10.105:8888', 'Content-Length': '32', 'Accept-Language': 'zh-CN,zh;q=0.8', 'Accept-Encoding': 'gzip,deflate,sdch', 'Host': '192.168.10.105:8888', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'User-Agent': 'Mozilla/5.0 (Windows NT 5.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/30.0.1581.2 Safari/537.36', 'Connection': 'keep-alive', 'Referer': 'http://192.168.10.105:8888/', 'Cache-Control': 'max-age=0', 'Content-Type': 'application/x-www-form-urlencoded'})
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1141, in _when_complete
        callback()
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1162, in _execute_method
        self._when_complete(method(*self.path_args, **self.path_kwargs),
      File "/home/jack/workspace/order/handlers/qhandler.py", line 54, in post
        fd = open(file_name)
    IOError: [Errno 2] No such file or directory: '/home/jack/workspace/order/data/2013-09-10'
[E 130910 16:09:36 web:1635] 500 POST / (192.168.10.148) 10.09ms
[I 130910 16:11:26 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 16:11:27 app:9] app start
[I 130910 16:11:31 web:1635] 304 GET / (192.168.10.148) 5.24ms
[E 130910 16:11:35 web:1228] Uncaught exception POST / (192.168.10.148)
    HTTPRequest(protocol='http', host='192.168.10.105:8888', method='POST', uri='/', version='HTTP/1.1', remote_ip='192.168.10.148', headers={'Origin': 'http://192.168.10.105:8888', 'Content-Length': '32', 'Accept-Language': 'zh-CN,zh;q=0.8', 'Accept-Encoding': 'gzip,deflate,sdch', 'Host': '192.168.10.105:8888', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'User-Agent': 'Mozilla/5.0 (Windows NT 5.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/30.0.1581.2 Safari/537.36', 'Connection': 'keep-alive', 'Referer': 'http://192.168.10.105:8888/', 'Cache-Control': 'max-age=0', 'Content-Type': 'application/x-www-form-urlencoded'})
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1141, in _when_complete
        callback()
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1162, in _execute_method
        self._when_complete(method(*self.path_args, **self.path_kwargs),
      File "/home/jack/workspace/order/handlers/qhandler.py", line 54, in post
        fd = open(file_name, "wt")
    IOError: [Errno 2] No such file or directory: '/home/jack/workspace/order/data/2013-09-10'
[E 130910 16:11:35 web:1635] 500 POST / (192.168.10.148) 11.56ms
[I 130910 16:12:00 web:1635] 304 GET / (192.168.10.148) 1.95ms
[I 130910 16:12:03 web:1635] 200 POST / (192.168.10.148) 2.72ms
[I 130910 16:12:56 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 16:12:56 app:9] app start
[I 130910 16:13:00 web:1635] 200 POST / (192.168.10.148) 6.31ms
[I 130910 16:14:28 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 16:14:28 app:9] app start
[I 130910 16:14:31 web:1635] 200 GET / (192.168.10.148) 5.24ms
[I 130910 16:14:34 web:1635] 200 POST / (192.168.10.148) 3.97ms
[I 130910 16:17:30 web:1635] 200 GET / (192.168.10.148) 1.59ms
[I 130910 16:17:32 web:1635] 304 GET / (192.168.10.148) 1.79ms
[I 130910 16:17:35 web:1635] 200 POST / (192.168.10.148) 3.68ms
[I 130910 16:19:36 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 16:19:36 app:9] app start
[I 130910 16:19:40 web:1635] 200 GET / (192.168.10.148) 5.39ms
[I 130910 16:19:45 web:1635] 200 POST / (192.168.10.148) 2.00ms
[I 130910 16:20:51 web:1635] 200 POST / (192.168.10.148) 3.33ms
[I 130910 16:21:29 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 16:21:29 app:9] app start
[I 130910 16:21:32 web:1635] 200 GET / (192.168.10.148) 5.17ms
[I 130910 16:21:37 web:1635] 200 POST / (192.168.10.148) 5.59ms
[I 130910 16:23:32 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 16:23:33 app:9] app start
[I 130910 16:23:40 web:1635] 200 POST / (192.168.10.148) 10.40ms
[I 130910 16:24:53 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 16:24:53 app:9] app start
[I 130910 16:24:58 web:1635] 200 GET / (192.168.10.148) 5.37ms
[I 130910 16:25:02 web:1635] 200 POST / (192.168.10.148) 125.17ms
[I 130910 16:25:40 web:1635] 200 POST / (192.168.10.148) 7.13ms
[I 130910 16:31:27 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 16:31:28 app:9] app start
[I 130910 16:33:24 app:9] app start
[I 130910 16:33:53 app:9] app start
[I 130910 16:33:56 web:1635] 200 GET / (192.168.10.148) 5.11ms
[I 130910 16:34:00 web:1635] 200 POST / (192.168.10.148) 7.33ms
[I 130910 16:34:01 web:1635] 200 GET /today (192.168.10.148) 4.22ms
[I 130910 16:34:17 web:1635] 200 GET / (192.168.10.148) 1.60ms
[I 130910 16:34:21 web:1635] 304 GET /today (192.168.10.148) 4.13ms
[I 130910 16:34:30 web:1635] 200 POST / (192.168.10.148) 6.11ms
[I 130910 16:34:32 web:1635] 200 GET /today (192.168.10.148) 6.77ms
[I 130910 16:34:38 web:1635] 200 POST / (192.168.10.148) 7.61ms
[I 130910 16:39:03 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 16:39:03 app:9] app start
[I 130910 16:39:07 web:1635] 200 GET / (192.168.10.148) 18.45ms
[I 130910 16:39:13 web:1635] 200 POST / (192.168.10.148) 9.73ms
[I 130910 16:39:34 web:1635] 200 POST / (192.168.10.148) 6.45ms
[I 130910 16:39:37 web:1635] 200 POST / (192.168.10.148) 7.74ms
[I 130910 16:39:38 web:1635] 200 GET /today (192.168.10.148) 5.29ms
[I 130910 16:50:28 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 16:50:28 app:9] app start
[I 130910 16:50:38 web:1635] 200 GET / (192.168.10.148) 5.84ms
[E 130910 16:50:42 web:1228] Uncaught exception POST / (192.168.10.148)
    HTTPRequest(protocol='http', host='192.168.10.105:8888', method='POST', uri='/', version='HTTP/1.1', remote_ip='192.168.10.148', headers={'Origin': 'http://192.168.10.105:8888', 'Content-Length': '32', 'Accept-Language': 'zh-CN,zh;q=0.8', 'Accept-Encoding': 'gzip,deflate,sdch', 'Host': '192.168.10.105:8888', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'User-Agent': 'Mozilla/5.0 (Windows NT 5.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/30.0.1581.2 Safari/537.36', 'Connection': 'keep-alive', 'Referer': 'http://192.168.10.105:8888/', 'Cache-Control': 'max-age=0', 'Content-Type': 'application/x-www-form-urlencoded'})
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1141, in _when_complete
        callback()
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1162, in _execute_method
        self._when_complete(method(*self.path_args, **self.path_kwargs),
      File "/home/jack/workspace/order/handlers/qhandler.py", line 74, in post
        fd.write(json.dumps(day_json, indent=1, ensure_ascii=False))
      File "/usr/lib/python2.7/json/__init__.py", line 238, in dumps
        **kw).encode(obj)
      File "/usr/lib/python2.7/json/encoder.py", line 204, in encode
        return ''.join(chunks)
    UnicodeDecodeError: 'ascii' codec can't decode byte 0xe7 in position 1: ordinal not in range(128)
[E 130910 16:50:42 web:1635] 500 POST / (192.168.10.148) 66.79ms
[I 130910 16:51:45 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 16:52:39 app:9] app start
[I 130910 16:52:42 web:1635] 304 GET / (192.168.10.148) 5.28ms
[E 130910 16:52:45 web:1228] Uncaught exception POST / (192.168.10.148)
    HTTPRequest(protocol='http', host='192.168.10.105:8888', method='POST', uri='/', version='HTTP/1.1', remote_ip='192.168.10.148', headers={'Origin': 'http://192.168.10.105:8888', 'Content-Length': '32', 'Accept-Language': 'zh-CN,zh;q=0.8', 'Accept-Encoding': 'gzip,deflate,sdch', 'Host': '192.168.10.105:8888', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'User-Agent': 'Mozilla/5.0 (Windows NT 5.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/30.0.1581.2 Safari/537.36', 'Connection': 'keep-alive', 'Referer': 'http://192.168.10.105:8888/', 'Cache-Control': 'max-age=0', 'Content-Type': 'application/x-www-form-urlencoded'})
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1141, in _when_complete
        callback()
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1162, in _execute_method
        self._when_complete(method(*self.path_args, **self.path_kwargs),
      File "/home/jack/workspace/order/handlers/qhandler.py", line 67, in post
        day_json = json.loads(json_str)
      File "/usr/lib/python2.7/json/__init__.py", line 326, in loads
        return _default_decoder.decode(s)
      File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
      File "/usr/lib/python2.7/json/decoder.py", line 384, in raw_decode
        raise ValueError("No JSON object could be decoded")
    ValueError: No JSON object could be decoded
[E 130910 16:52:45 web:1635] 500 POST / (192.168.10.148) 39.71ms
[I 130910 16:53:22 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 16:53:22 app:9] app start
[I 130910 16:53:24 web:1635] 304 GET / (192.168.10.148) 5.59ms
[E 130910 16:53:28 web:1228] Uncaught exception POST / (192.168.10.148)
    HTTPRequest(protocol='http', host='192.168.10.105:8888', method='POST', uri='/', version='HTTP/1.1', remote_ip='192.168.10.148', headers={'Origin': 'http://192.168.10.105:8888', 'Content-Length': '32', 'Accept-Language': 'zh-CN,zh;q=0.8', 'Accept-Encoding': 'gzip,deflate,sdch', 'Host': '192.168.10.105:8888', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'User-Agent': 'Mozilla/5.0 (Windows NT 5.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/30.0.1581.2 Safari/537.36', 'Connection': 'keep-alive', 'Referer': 'http://192.168.10.105:8888/', 'Cache-Control': 'max-age=0', 'Content-Type': 'application/x-www-form-urlencoded'})
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1141, in _when_complete
        callback()
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1162, in _execute_method
        self._when_complete(method(*self.path_args, **self.path_kwargs),
      File "/home/jack/workspace/order/handlers/qhandler.py", line 67, in post
        day_json = json.loads(json_str.encode("utf-8"))
      File "/usr/lib/python2.7/json/__init__.py", line 326, in loads
        return _default_decoder.decode(s)
      File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
      File "/usr/lib/python2.7/json/decoder.py", line 384, in raw_decode
        raise ValueError("No JSON object could be decoded")
    ValueError: No JSON object could be decoded
[E 130910 16:53:28 web:1635] 500 POST / (192.168.10.148) 16.72ms
[I 130910 16:53:36 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 16:53:36 app:9] app start
[I 130910 16:53:38 web:1635] 304 GET / (192.168.10.148) 5.25ms
[E 130910 16:53:41 web:1228] Uncaught exception POST / (192.168.10.148)
    HTTPRequest(protocol='http', host='192.168.10.105:8888', method='POST', uri='/', version='HTTP/1.1', remote_ip='192.168.10.148', headers={'Origin': 'http://192.168.10.105:8888', 'Content-Length': '32', 'Accept-Language': 'zh-CN,zh;q=0.8', 'Accept-Encoding': 'gzip,deflate,sdch', 'Host': '192.168.10.105:8888', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'User-Agent': 'Mozilla/5.0 (Windows NT 5.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/30.0.1581.2 Safari/537.36', 'Connection': 'keep-alive', 'Referer': 'http://192.168.10.105:8888/', 'Cache-Control': 'max-age=0', 'Content-Type': 'application/x-www-form-urlencoded'})
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1141, in _when_complete
        callback()
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1162, in _execute_method
        self._when_complete(method(*self.path_args, **self.path_kwargs),
      File "/home/jack/workspace/order/handlers/qhandler.py", line 67, in post
        day_json = json.loads(json_str.decode("utf-8"))
      File "/usr/lib/python2.7/json/__init__.py", line 326, in loads
        return _default_decoder.decode(s)
      File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
      File "/usr/lib/python2.7/json/decoder.py", line 384, in raw_decode
        raise ValueError("No JSON object could be decoded")
    ValueError: No JSON object could be decoded
[E 130910 16:53:41 web:1635] 500 POST / (192.168.10.148) 16.67ms
[I 130910 16:54:24 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 16:54:25 app:9] app start
[E 130910 16:54:28 web:1228] Uncaught exception POST / (192.168.10.148)
    HTTPRequest(protocol='http', host='192.168.10.105:8888', method='POST', uri='/', version='HTTP/1.1', remote_ip='192.168.10.148', headers={'Origin': 'http://192.168.10.105:8888', 'Content-Length': '32', 'Accept-Language': 'zh-CN,zh;q=0.8', 'Accept-Encoding': 'gzip,deflate,sdch', 'Host': '192.168.10.105:8888', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'User-Agent': 'Mozilla/5.0 (Windows NT 5.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/30.0.1581.2 Safari/537.36', 'Connection': 'keep-alive', 'Referer': 'http://192.168.10.105:8888/', 'Cache-Control': 'max-age=0', 'Content-Type': 'application/x-www-form-urlencoded'})
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1141, in _when_complete
        callback()
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1162, in _execute_method
        self._when_complete(method(*self.path_args, **self.path_kwargs),
      File "/home/jack/workspace/order/handlers/qhandler.py", line 67, in post
        day_json = json.loads(json_str)#.decode("utf-8"))
      File "/usr/lib/python2.7/json/__init__.py", line 326, in loads
        return _default_decoder.decode(s)
      File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
      File "/usr/lib/python2.7/json/decoder.py", line 384, in raw_decode
        raise ValueError("No JSON object could be decoded")
    ValueError: No JSON object could be decoded
[E 130910 16:54:28 web:1635] 500 POST / (192.168.10.148) 16.03ms
[I 130910 16:54:28 web:1635] 304 GET /favicon.ico (192.168.10.148) 5.37ms
[I 130910 16:54:38 web:1635] 304 GET / (192.168.10.148) 6.06ms
[E 130910 16:54:41 web:1228] Uncaught exception POST / (192.168.10.148)
    HTTPRequest(protocol='http', host='192.168.10.105:8888', method='POST', uri='/', version='HTTP/1.1', remote_ip='192.168.10.148', headers={'Origin': 'http://192.168.10.105:8888', 'Content-Length': '32', 'Accept-Language': 'zh-CN,zh;q=0.8', 'Accept-Encoding': 'gzip,deflate,sdch', 'Host': '192.168.10.105:8888', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'User-Agent': 'Mozilla/5.0 (Windows NT 5.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/30.0.1581.2 Safari/537.36', 'Connection': 'keep-alive', 'Referer': 'http://192.168.10.105:8888/', 'Cache-Control': 'max-age=0', 'Content-Type': 'application/x-www-form-urlencoded'})
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1141, in _when_complete
        callback()
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1162, in _execute_method
        self._when_complete(method(*self.path_args, **self.path_kwargs),
      File "/home/jack/workspace/order/handlers/qhandler.py", line 67, in post
        day_json = json.loads(json_str)#.decode("utf-8"))
      File "/usr/lib/python2.7/json/__init__.py", line 326, in loads
        return _default_decoder.decode(s)
      File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
      File "/usr/lib/python2.7/json/decoder.py", line 384, in raw_decode
        raise ValueError("No JSON object could be decoded")
    ValueError: No JSON object could be decoded
[E 130910 16:54:41 web:1635] 500 POST / (192.168.10.148) 16.30ms
[I 130910 16:58:38 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 16:58:39 app:9] app start
[I 130910 16:58:41 web:1635] 304 GET / (192.168.10.148) 5.90ms
[E 130910 16:58:45 web:1228] Uncaught exception POST / (192.168.10.148)
    HTTPRequest(protocol='http', host='192.168.10.105:8888', method='POST', uri='/', version='HTTP/1.1', remote_ip='192.168.10.148', headers={'Origin': 'http://192.168.10.105:8888', 'Content-Length': '32', 'Accept-Language': 'zh-CN,zh;q=0.8', 'Accept-Encoding': 'gzip,deflate,sdch', 'Host': '192.168.10.105:8888', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'User-Agent': 'Mozilla/5.0 (Windows NT 5.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/30.0.1581.2 Safari/537.36', 'Connection': 'keep-alive', 'Referer': 'http://192.168.10.105:8888/', 'Cache-Control': 'max-age=0', 'Content-Type': 'application/x-www-form-urlencoded'})
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1141, in _when_complete
        callback()
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1162, in _execute_method
        self._when_complete(method(*self.path_args, **self.path_kwargs),
      File "/home/jack/workspace/order/handlers/qhandler.py", line 67, in post
        day_json = json.loads(json_str, encoding="utf-8")#.decode("utf-8"))
      File "/usr/lib/python2.7/json/__init__.py", line 339, in loads
        return cls(encoding=encoding, **kw).decode(s)
      File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
      File "/usr/lib/python2.7/json/decoder.py", line 384, in raw_decode
        raise ValueError("No JSON object could be decoded")
    ValueError: No JSON object could be decoded
[E 130910 16:58:45 web:1635] 500 POST / (192.168.10.148) 18.51ms
[I 130910 16:59:36 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 16:59:36 app:9] app start
[I 130910 16:59:57 web:1635] 304 GET / (192.168.10.148) 6.20ms
[I 130910 17:00:00 web:1635] 200 POST / (192.168.10.148) 5.37ms
[I 130910 17:00:10 web:1635] 200 GET /today (192.168.10.148) 5.33ms
[I 130910 17:00:13 web:1635] 200 POST / (192.168.10.148) 7.27ms
[E 130910 17:00:34 web:1228] Uncaught exception GET /today (192.168.10.148)
    HTTPRequest(protocol='http', host='192.168.10.105:8888', method='GET', uri='/today', version='HTTP/1.1', remote_ip='192.168.10.148', headers={'Accept-Language': 'zh-CN,zh;q=0.8', 'Accept-Encoding': 'gzip,deflate,sdch', 'Host': '192.168.10.105:8888', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'User-Agent': 'Mozilla/5.0 (Windows NT 5.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/30.0.1581.2 Safari/537.36', 'Connection': 'keep-alive', 'Referer': 'http://192.168.10.105:8888/', 'If-None-Match': '"1744c984d9b25cff989033cf45e90efc23214670"'})
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1141, in _when_complete
        callback()
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1162, in _execute_method
        self._when_complete(method(*self.path_args, **self.path_kwargs),
      File "/home/jack/workspace/order/handlers/qhandler.py", line 50, in get
        day_json = json.loads(json_str)
      File "/usr/lib/python2.7/json/__init__.py", line 326, in loads
        return _default_decoder.decode(s)
      File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
      File "/usr/lib/python2.7/json/decoder.py", line 384, in raw_decode
        raise ValueError("No JSON object could be decoded")
    ValueError: No JSON object could be decoded
[E 130910 17:00:34 web:1635] 500 GET /today (192.168.10.148) 13.33ms
[I 130910 17:00:52 web:1635] 200 GET / (192.168.10.148) 2.29ms
[E 130910 17:00:53 web:1228] Uncaught exception GET /today (192.168.10.148)
    HTTPRequest(protocol='http', host='192.168.10.105:8888', method='GET', uri='/today', version='HTTP/1.1', remote_ip='192.168.10.148', headers={'Connection': 'keep-alive', 'Accept-Language': 'zh-CN,zh;q=0.8', 'Accept-Encoding': 'gzip,deflate,sdch', 'Referer': 'http://192.168.10.105:8888/', 'Host': '192.168.10.105:8888', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'User-Agent': 'Mozilla/5.0 (Windows NT 5.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/30.0.1581.2 Safari/537.36'})
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1141, in _when_complete
        callback()
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1162, in _execute_method
        self._when_complete(method(*self.path_args, **self.path_kwargs),
      File "/home/jack/workspace/order/handlers/qhandler.py", line 50, in get
        day_json = json.loads(json_str)
      File "/usr/lib/python2.7/json/__init__.py", line 326, in loads
        return _default_decoder.decode(s)
      File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
      File "/usr/lib/python2.7/json/decoder.py", line 382, in raw_decode
        obj, end = self.scan_once(s, idx)
    ValueError: Expecting property name: line 2 column 2 (char 3)
[E 130910 17:00:53 web:1635] 500 GET /today (192.168.10.148) 17.59ms
[I 130910 17:01:52 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 17:01:52 app:9] app start
[I 130910 17:01:56 web:1635] 304 GET / (192.168.10.148) 5.55ms
[I 130910 17:01:56 web:1635] 304 GET / (192.168.10.148) 2.21ms
[I 130910 17:01:59 web:1635] 200 POST / (192.168.10.148) 4.48ms
[I 130910 17:02:07 web:1635] 200 GET /today (192.168.10.148) 3.99ms
[I 130910 17:02:11 web:1635] 200 POST / (192.168.10.148) 6.76ms
[I 130910 17:02:21 web:1635] 200 POST / (192.168.10.148) 5.69ms
[I 130910 17:02:30 web:1635] 200 POST / (192.168.10.148) 7.41ms
[I 130910 17:02:35 web:1635] 200 POST / (192.168.10.148) 7.34ms
[I 130910 17:02:38 web:1635] 200 POST / (192.168.10.148) 6.54ms
[I 130910 17:02:45 web:1635] 200 POST / (192.168.10.148) 6.07ms
[I 130910 17:03:00 web:1635] 200 POST / (192.168.10.148) 7.34ms
[I 130910 17:03:07 web:1635] 200 GET /today (192.168.10.148) 5.84ms
[I 130910 17:03:09 web:1635] 200 POST / (192.168.10.148) 9.87ms
[I 130910 17:04:08 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 17:04:08 app:9] app start
[I 130910 17:04:21 web:1635] 200 POST / (192.168.10.148) 12.24ms
[I 130910 17:04:22 web:1635] 200 GET /today (192.168.10.148) 6.28ms
[I 130910 17:06:35 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 17:06:35 app:9] app start
[I 130910 17:06:37 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 17:06:38 app:9] app start
[I 130910 17:10:04 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 17:25:03 app:9] app start
[I 130910 17:25:07 web:1635] 200 GET / (192.168.10.148) 5.99ms
[I 130910 17:25:11 web:1635] 200 POST / (192.168.10.148) 3.00ms
[I 130910 17:25:13 web:1635] 200 GET /today (192.168.10.148) 1.88ms
[I 130910 17:25:37 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 17:25:37 app:9] app start
[I 130910 17:27:10 web:1635] 200 GET / (192.168.10.148) 4.77ms
[I 130910 17:27:12 web:1635] 200 GET /del (192.168.10.148) 1.37ms
[I 130910 17:27:16 web:1635] 200 POST /del (192.168.10.148) 1.93ms
[I 130910 17:27:18 web:1635] 200 GET /del (192.168.10.148) 2.20ms
[I 130910 17:27:21 web:1635] 200 GET /today (192.168.10.148) 1.77ms
[I 130910 17:27:23 web:1635] 304 GET / (192.168.10.148) 1.65ms
[I 130910 17:27:25 web:1635] 304 GET /today (192.168.10.148) 1.98ms
[I 130910 17:27:29 web:1635] 304 GET /del (192.168.10.148) 1.40ms
[I 130910 17:27:33 web:1635] 200 POST /del (192.168.10.148) 1.48ms
[I 130910 17:27:36 web:1635] 304 GET / (192.168.10.148) 1.37ms
[I 130910 17:27:36 web:1635] 304 GET / (192.168.10.148) 1.23ms
[I 130910 17:27:38 web:1635] 200 GET /today (192.168.10.148) 2.02ms
[I 130910 17:28:21 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 17:28:22 app:9] app start
[I 130910 17:28:29 web:1635] 304 GET /today (192.168.10.148) 6.14ms
[I 130910 17:28:31 web:1635] 304 GET / (192.168.10.148) 1.82ms
[I 130910 17:28:35 web:1635] 200 POST / (192.168.10.148) 2.86ms
[I 130910 17:28:38 web:1635] 200 GET /today (192.168.10.148) 1.81ms
[I 130910 17:28:44 web:1635] 200 GET /del (192.168.10.148) 1.49ms
[E 130910 17:28:46 web:1228] Uncaught exception POST /del (192.168.10.148)
    HTTPRequest(protocol='http', host='192.168.10.105:8888', method='POST', uri='/del', version='HTTP/1.1', remote_ip='192.168.10.148', headers={'Origin': 'http://192.168.10.105:8888', 'Content-Length': '29', 'Accept-Language': 'zh-CN,zh;q=0.8', 'Accept-Encoding': 'gzip,deflate,sdch', 'Host': '192.168.10.105:8888', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'User-Agent': 'Mozilla/5.0 (Windows NT 5.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/30.0.1581.2 Safari/537.36', 'Connection': 'keep-alive', 'Referer': 'http://192.168.10.105:8888/del', 'Cache-Control': 'max-age=0', 'Content-Type': 'application/x-www-form-urlencoded'})
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1141, in _when_complete
        callback()
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1162, in _execute_method
        self._when_complete(method(*self.path_args, **self.path_kwargs),
      File "/home/jack/workspace/order/handlers/qhandler.py", line 79, in post
        self.render("today.html")
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 538, in render
        html = self.render_string(template_name, **kwargs)
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 645, in render_string
        return t.generate(**namespace)
      File "/usr/local/lib/python2.7/dist-packages/tornado/template.py", line 273, in generate
        return execute()
      File "today_html.generated.py", line 5, in _tt_execute
        _tt_tmp = day_json  # today.html:14
    NameError: global name 'day_json' is not defined
[E 130910 17:28:46 web:1635] 500 POST /del (192.168.10.148) 80.13ms
[I 130910 17:29:04 web:1635] 304 GET /del (192.168.10.148) 2.23ms
[I 130910 17:29:06 web:1635] 200 GET /today (192.168.10.148) 1.70ms
[I 130910 17:31:42 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 17:31:42 app:9] app start
[I 130910 17:31:46 web:1635] 304 GET /del (192.168.10.148) 5.41ms
[I 130910 17:31:49 web:1635] 302 POST /del (192.168.10.148) 2.48ms
[I 130910 17:31:49 web:1635] 304 GET /today (192.168.10.148) 1.92ms
[I 130910 17:31:53 web:1635] 200 GET / (192.168.10.148) 1.31ms
[I 130910 17:31:57 web:1635] 200 POST / (192.168.10.148) 3.11ms
[I 130910 17:31:59 web:1635] 200 GET /today (192.168.10.148) 3.52ms
[I 130910 17:32:03 web:1635] 200 POST / (192.168.10.148) 2.61ms
[I 130910 17:32:06 web:1635] 200 GET /del (192.168.10.148) 1.40ms
[I 130910 17:32:17 web:1635] 304 GET /today (192.168.10.148) 1.76ms
[I 130910 17:32:19 web:1635] 304 GET /del (192.168.10.148) 1.87ms
[I 130910 17:32:21 web:1635] 302 POST /del (192.168.10.148) 1.61ms
[I 130910 17:32:21 web:1635] 304 GET /today (192.168.10.148) 2.03ms
[I 130910 17:32:30 web:1635] 200 GET /del (192.168.10.148) 1.32ms
[I 130910 17:32:32 web:1635] 302 POST /del (192.168.10.148) 1.48ms
[I 130910 17:32:32 web:1635] 200 GET /today (192.168.10.148) 2.09ms
[I 130910 17:33:47 web:1635] 200 GET / (192.168.10.143) 2.24ms
[I 130910 17:33:47 web:1635] 200 GET /favicon.ico (192.168.10.143) 5.79ms
[I 130910 17:34:15 web:1635] 200 POST / (192.168.10.143) 3.13ms
[I 130910 17:34:16 web:1635] 200 GET /today (192.168.10.143) 2.15ms
[I 130910 17:34:23 web:1635] 200 GET /del (192.168.10.143) 1.27ms
[I 130910 17:34:42 web:1635] 302 POST /del (192.168.10.143) 1.66ms
[I 130910 17:34:42 web:1635] 200 GET /today (192.168.10.143) 1.79ms
[I 130910 17:34:44 web:1635] 304 GET / (192.168.10.143) 1.24ms
[I 130910 17:34:56 web:1635] 200 POST / (192.168.10.143) 2.99ms
[I 130910 17:34:58 web:1635] 200 GET /today (192.168.10.143) 1.76ms
[I 130910 17:35:45 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 17:35:45 app:9] app start
[I 130910 17:38:48 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 17:38:48 app:9] app start
[I 130910 17:46:13 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 17:46:13 app:9] app start
[I 130910 17:46:18 web:1635] 200 GET / (192.168.10.148) 4.79ms
[I 130910 17:46:18 web:1635] 304 GET / (192.168.10.148) 1.57ms
[I 130910 17:46:22 web:1635] 200 POST / (192.168.10.148) 2.20ms
[I 130910 17:46:23 web:1635] 200 GET /today (192.168.10.148) 2.12ms
[I 130910 17:46:33 web:1635] 200 POST / (192.168.10.148) 3.02ms
[I 130910 17:46:33 web:1635] 200 GET /today (192.168.10.148) 1.76ms
[I 130910 17:46:36 web:1635] 200 GET / (192.168.10.148) 1.20ms
[I 130910 17:46:41 web:1635] 200 POST / (192.168.10.148) 2.90ms
[I 130910 17:46:41 web:1635] 200 GET /today (192.168.10.148) 1.74ms
[I 130910 17:46:49 web:1635] 200 GET /del (192.168.10.148) 1.53ms
[I 130910 17:46:52 web:1635] 302 POST /del (192.168.10.148) 3.03ms
[I 130910 17:46:52 web:1635] 200 GET /today (192.168.10.148) 2.75ms
[I 130910 17:47:12 web:1635] 200 GET / (192.168.10.148) 1.36ms
[I 130910 17:47:15 web:1635] 200 GET /del (192.168.10.148) 2.21ms
[I 130910 17:47:18 web:1635] 302 POST /del (192.168.10.148) 3.09ms
[I 130910 17:47:18 web:1635] 200 GET /today (192.168.10.148) 2.22ms
[I 130910 17:47:20 web:1635] 200 GET /del (192.168.10.148) 1.57ms
[I 130910 17:47:24 web:1635] 302 POST /del (192.168.10.148) 3.74ms
[I 130910 17:47:24 web:1635] 200 GET /today (192.168.10.148) 1.78ms
[I 130910 17:48:08 web:1635] 304 GET / (192.168.10.143) 1.66ms
[I 130910 17:48:28 web:1635] 200 POST / (192.168.10.143) 2.81ms
[I 130910 17:48:30 web:1635] 200 GET /today (192.168.10.143) 1.96ms
[I 130910 17:48:35 web:1635] 304 GET / (192.168.10.143) 1.29ms
[I 130910 17:48:45 web:1635] 200 POST / (192.168.10.143) 3.39ms
[I 130910 17:48:57 web:1635] 200 POST / (192.168.10.143) 2.59ms
[I 130910 17:48:58 web:1635] 200 GET /today (192.168.10.143) 1.86ms
[I 130910 17:49:24 web:1635] 304 GET / (192.168.10.143) 1.31ms
[I 130910 17:49:33 web:1635] 200 POST / (192.168.10.143) 3.17ms
[I 130910 17:49:34 web:1635] 200 GET /today (192.168.10.143) 1.82ms
[I 130910 17:50:20 web:1635] 304 GET / (192.168.10.143) 1.45ms
[I 130910 17:51:14 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 17:51:15 app:9] app start
[I 130910 17:51:26 web:1635] 200 GET /today (192.168.10.148) 6.42ms
[I 130910 17:51:26 web:1635] 304 GET /favicon.ico (192.168.10.148) 6.25ms
[I 130910 17:51:31 web:1635] 200 GET /del (192.168.10.148) 1.28ms
[I 130910 17:51:35 web:1635] 302 POST /del (192.168.10.148) 1.97ms
[I 130910 17:51:35 web:1635] 304 GET /today (192.168.10.148) 2.26ms
[I 130910 17:51:39 web:1635] 304 GET / (192.168.10.148) 1.42ms
[I 130910 17:51:43 web:1635] 200 POST / (192.168.10.148) 2.85ms
[I 130910 17:51:43 web:1635] 200 GET /del (192.168.10.148) 1.66ms
[I 130910 17:51:50 web:1635] 200 GET /today (192.168.10.148) 2.11ms
[I 130910 17:51:55 web:1635] 200 GET / (192.168.10.148) 1.22ms
[I 130910 17:51:56 web:1635] 304 GET /del (192.168.10.148) 1.18ms
[E 130910 17:51:59 web:1228] Uncaught exception POST /del (192.168.10.148)
    HTTPRequest(protocol='http', host='192.168.10.105:8888', method='POST', uri='/del', version='HTTP/1.1', remote_ip='192.168.10.148', headers={'Origin': 'http://192.168.10.105:8888', 'Content-Length': '8', 'Accept-Language': 'zh-CN,zh;q=0.8', 'Accept-Encoding': 'gzip,deflate,sdch', 'Host': '192.168.10.105:8888', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'User-Agent': 'Mozilla/5.0 (Windows NT 5.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/30.0.1581.2 Safari/537.36', 'Connection': 'keep-alive', 'Referer': 'http://192.168.10.105:8888/del', 'Cache-Control': 'max-age=0', 'Content-Type': 'application/x-www-form-urlencoded'})
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1141, in _when_complete
        callback()
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1162, in _execute_method
        self._when_complete(method(*self.path_args, **self.path_kwargs),
      File "/home/jack/workspace/order/handlers/qhandler.py", line 76, in post
        for c in j:
    RuntimeError: dictionary changed size during iteration
[E 130910 17:51:59 web:1635] 500 POST /del (192.168.10.148) 8.38ms
[I 130910 17:53:16 autoreload:186] /home/jack/workspace/order/handlers/qhandler.py modified; restarting server
[I 130910 17:53:16 app:9] app start
[I 130910 17:53:21 web:1635] 304 GET / (192.168.10.148) 5.32ms
[I 130910 17:53:22 web:1635] 304 GET /today (192.168.10.148) 2.39ms
[I 130910 17:53:29 web:1635] 304 GET /del (192.168.10.148) 1.24ms
[I 130910 17:53:31 web:1635] 302 POST /del (192.168.10.148) 2.33ms
[I 130910 17:53:31 web:1635] 200 GET /today (192.168.10.148) 1.98ms
[I 130911 09:13:20 app:9] app start
[I 130911 09:15:49 app:9] app start
[I 130911 09:16:02 web:1635] 200 GET / (192.168.10.148) 17.61ms
[I 130911 09:16:07 web:1635] 200 POST / (192.168.10.148) 4.50ms
[I 130911 09:16:09 web:1635] 200 GET /today (192.168.10.148) 15.31ms
[I 130911 09:16:13 web:1635] 200 POST / (192.168.10.148) 3.11ms
